docker-compose.yml

```diff
@@ -19,7 +19,7 @@ services:
       POSTGRES_PASSWORD: password
       POSTGRES_DB: sif
     ports:
-      - 5432
+      - 5432:5432
```

Comment:
is this intended? (not too familiar with myself)

Reply:
I think one port is the port inside the container and the other is outside the container (or the other way around)

Reply:
Yes! This lets the postgres docker image be useable from your host machine. 

So if you're comfortable with docker, you could run `docker compose up -d db` from this project directory and have Postgres without having to install it on your OS directly.

Reply:
in docker configuration--ports and volumes in particular--the colon-separated `thing:thing` refers to `outside:inside` or `host:image`. 

In this case, `5432:5432` read left to right says, "expose on port 5432 on my host machine the port 5432 from inside the docker container". 

When we use a single value like `5432`, it means "the docker container I'm describing should expose port 5432 to other containers on the same network." "The same network" here means the other services named in docker-compose.yml.